### PR TITLE
BF: catch possible crashes  in except handlers + few more minor fixes

### PR DESCRIPTION
**Pull request by yarikoptic**
while playing "in real-time"

sorry -- no dedicated tests... getting late here.  Just played around with install and get while providing bogus targets ;)

### CODE REVIEW

**Comment by yarikoptic** [comment]
Review/improvements anyone, or merge?

**Comment by bpoldrack** [comment]
Only minor remark, so merging ...

**Comment by mih** [comment]
Merge monkey says: I got a 0.5% coverage drop for free (cumulative).....

File: datalad/distribution/install.py [hunk 1]
```diff
@@ -255,7 +255,9 @@ def __call__(
                         **common_kwargs
                     )
                 except IncompleteResultsError as exc:
-                    lgr.warning("Some items failed to get: %s", exc_str(exc))
+                    exc_str_ = ': ' + exc_str(exc) if exc.results else ''
+                    lgr.warning("Some items failed to get%s",
```
Comment (yarikoptic) [inline datalad/distribution/install.py]: Oh cool github, did it right on the phone

**Comment by yarikoptic** [comment]
Yes, we still lack many cases unittest s

**Pull request merged**